pkg/auth/dependency/provider/password/mock_password.go

```diff
@@ -40,6 +41,19 @@ func (m *MockProvider) IsAuthDataMatching(authData map[string]string) bool {
 	return m.authDataChecker.isMatching(authData)
 }
 
+func (m *MockProvider) GetLoginIDMetadataFlattenedKeys() []string {
+	var output []string
+	for _, keys := range m.loginIDMetadataKeys {
+		for _, key := range keys {
+			if !utils.SliceContains(output, key) {
```

Comment:
This ends up with a 3-level nested loop, which the last layer is not needed.

Something like this: https://kylewbanks.com/blog/creating-unique-slices-in-go